File: docs/DSA/neetcode150/src/py/group_anagrams.py
from typing import List
class Solution:
    def _get_bitmap(self, s: str) -> tuple:
        l = [0] * 26
        for ch in s:
            l[ord(ch) - ord('a')] += 1
        return tuple(l)

    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        freq = {}  # bitmap -> str

        for s in strs:
            bitmap = self._get_bitmap(s)
            if bitmap in freq:
                freq[bitmap].append(s)
            else:
                freq[bitmap] = [s]
        return list(freq.values())

File: docs/DSA/neetcode150/src/go/group_anagrams.go
package main

func getBitmap(s string) [26]int {
	var bitmap [26]int
	for _, ch := range s {
		bitmap[ch-'a']++
	}
	return bitmap
}

func groupAnagrams(strs []string) [][]string {
	freq := make(map[[26]int][]string)
	for _, str := range strs {
		bitmap := getBitmap(str)
		freq[bitmap] = append(freq[bitmap], str)
	}

	var groups [][]string
	for _, group := range freq {
		groups = append(groups, group)
	}
	return groups
}
